homeassistant/components/lg_thinq/number.py

```diff
@@ -0,0 +1,206 @@
+"""Support for number entities."""
+
+from __future__ import annotations
+
+import logging
+
+from thinqconnect import DeviceType
+from thinqconnect.devices.const import Property as ThinQProperty
+from thinqconnect.integration import ActiveMode, TimerProperty
+
+from homeassistant.components.number import (
+    NumberDeviceClass,
+    NumberEntity,
+    NumberEntityDescription,
+    NumberMode,
+)
+from homeassistant.const import PERCENTAGE, UnitOfTemperature, UnitOfTime
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import ThinqConfigEntry
+from .entity import ThinQEntity
+
+NUMBER_DESC: dict[ThinQProperty, NumberEntityDescription] = {
+    ThinQProperty.FAN_SPEED: NumberEntityDescription(
+        key=ThinQProperty.FAN_SPEED,
+        translation_key=ThinQProperty.FAN_SPEED,
+    ),
+    ThinQProperty.LAMP_BRIGHTNESS: NumberEntityDescription(
+        key=ThinQProperty.LAMP_BRIGHTNESS,
+        translation_key=ThinQProperty.LAMP_BRIGHTNESS,
+    ),
```

Comment:
unit of measurement?

Reply:
It is level value.
Fan speed : 0~2
Brightness : 0~4

Reply:
I might think that having this as a select entity would be more useful, but I don't think it's a blocker

Reply:
Because their type is range(has min/max/step) we want to use number entity.